src/sentry/monitors/system_incidents.py

```diff
@@ -166,6 +177,258 @@ def get_clock_tick_volume_metric(tick: datetime) -> float | None:
+    tick_metrics = [float(value) if value else None for value in histories[:-1]]
+
+    # The last decision is the last value fetched
+    if histories[-1] is not None:
+        last_decision = Decision.from_str(histories[-1])
+    else:
+        # By default the previous decision is used. If there was no previous
+        # decision we can only assume things are operating normally
+        last_decision = Decision.NORMAL
+
+    # Pre-compute decisions for the history of metrics
+    metric_decisions = [_metric_decision(metric) for metric in tick_metrics]
+
+    def make_decision(
+        decision: TickAnomalyDecision,
+        transition: AnomalyTransition | None = None,
+    ) -> DecisionResult:
+        decision_key = MONITOR_TICK_DECISION.format(ts=_make_reference_ts(tick))
+        redis_client.set(decision_key, decision)
+        redis_client.expire(decision_key, MONITOR_VOLUME_RETENTION)
+
+        return DecisionResult(decision, transition)
+
+    def metric_matches(decision: TickAnomalyDecision) -> Generator[bool]:
+        return (d == decision for d in metric_decisions)
+
+    # ENTERING an abnormality
+    #
+    # If we've detected an anomaly and we're not already in an incident or
+    # anomalous state, mark this tick as anomalous.
+    if last_decision == Decision.NORMAL and metric_decisions[-1] == Decision.ABNORMAL:
+        return make_decision(Decision.ABNORMAL, AnomalyTransition.ABNORMALITY_STARTED)
+
+    # RECOVERED from abnormality
+    #
+    # If the previous result was anomalous check and if we have recovered and can
+    # backfill these decisions as normal
+    if last_decision == Decision.ABNORMAL and all(metric_matches(Decision.NORMAL)):
+        _backfill_decisions(past_ts, Decision.NORMAL, Decision.ABNORMAL)
+        return make_decision(Decision.NORMAL, AnomalyTransition.ABNORMALITY_RECOVERED)
+
+    # RECOVERING from an incident
+    #
+    # If we are actively in an incident and the most recent metric decision has
+    # recovered to normal we can de-escalate the incident to abnormal.
+    if last_decision == Decision.INCIDENT and metric_decisions[-1] == Decision.NORMAL:
+        return make_decision(Decision.RECOVERING, AnomalyTransition.INCIDENT_RECOVERING)
+
+    # RECOVERED from an incident
+    #
+    # If the previous result was anomalous check if we have recovered and can
+    # backfill these decisions as normal
+    if last_decision == Decision.RECOVERING and all(metric_matches(Decision.NORMAL)):
+        _backfill_decisions(past_ts, Decision.NORMAL, Decision.RECOVERING)
+        return make_decision(Decision.NORMAL, AnomalyTransition.INCIDENT_RECOVERED)
+
+    # FAILED RECOVERY from an incident
+    #
+    # If an incident had begun recovering but we've detected an abnormality
+    # again backfill all recovery decisions to an incident decision.
+    if last_decision == Decision.RECOVERING and metric_decisions[-1] == Decision.ABNORMAL:
```

Comment:
should this check if last metric decision is incident?

Reply:
Yes. this was bug. Fixed